pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/SumPrecisionAggregationFunction.java

```diff
@@ -108,6 +108,7 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
           sum = sum.add(new BigDecimal(stringValues[i]));
         }
         break;
+      case BIG_DECIMAL:
```

Comment:
(MAJOR) We should be able to directly read big decimals (that's the main point of this PR). We should also modify `aggregateGroupBySV` and `aggregateGroupByMV` to handle `BIG_DECIMAL` type